solana/programs/ntt-quoter/src/processor/request_relay.rs

```diff
@@ -0,0 +1,138 @@
+pub struct RequestRelay<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    #[account(seeds = [Instance::SEED_PREFIX], bump = Instance::BUMP)]
+    pub instance: Account<'info, Instance>,
+
+    #[account(mut, address = instance.fee_recipient @ NttQuoterError::InvalidFeeRecipient)]
+    /// CHECK: leave britney alone (anchor complains about a missing check despite the address check)
+    pub fee_recipient: AccountInfo<'info>,
+
+    #[account(
+    seeds = [RegisteredChain::SEED_PREFIX, outbox_item.recipient_chain.id.to_be_bytes().as_ref()],
+    bump = registered_chain.bump,
+    constraint = registered_chain.base_price != u64::MAX @ NttQuoterError::RelayingToChainDisabled
+  )]
+    pub registered_chain: Account<'info, RegisteredChain>,
+
+    //TODO eventually drop the released constraint and instead implement release by relayer
+    #[account(
+    owner = example_native_token_transfers::ID,
+    constraint = outbox_item.released.get(WORMHOLE_TRANSCEIVER_INDEX),
+  )]
+    pub outbox_item: Account<'info, OutboxItem>,
+
+    //TODO should this be init_if_needed?
+    //     are there any security considerations here?
+    //     init_if_needed would allow for multiple requests, even after it was already delivered
+    //     right now requesting a relay for a request is permissionless which would allow
+    //       a third party to initiate a relay without gas dropoff for the user (though they
+    //       can just use connect and get a gas dropoff to the same wallet that way)
+    #[account(
+    init,
+    payer = payer,
+    space = 8 + RelayRequest::INIT_SPACE,
+    seeds = [RelayRequest::SEED_PREFIX, outbox_item.key().as_ref()],
+    bump,
+  )]
+    pub relay_request: Account<'info, RelayRequest>,
+
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct RequestRelayArgs {
+    pub gas_dropoff: u64, //NativeAmount,
+    pub max_fee: u64,     //SolAmount,
+}
+
+const GWEI: u64 = u64::pow(10, 9);
+
+fn mul_div(scalar: u64, numerator: u64, denominator: u64) -> u64 {
+    ((scalar as u128) * (numerator as u128) / (denominator as u128))
+        .try_into()
+        .unwrap()
+}
+
+//TODO built-in u128 division likely still wastes a ton of compute units
+//     might be more efficient to use f64 or ruint crate
+pub fn request_relay(ctx: Context<RequestRelay>, args: RequestRelayArgs) -> Result<()> {
+    let accs = ctx.accounts;
+
+    require_gt!(
+        args.gas_dropoff,
+        accs.registered_chain.max_gas_dropoff,
+        NttQuoterError::ExceedsMaxGasDropoff
+    );
+
+    let total_fee_in_lamports_without_rent = {
+        let target_native_in_gwei = args.gas_dropoff
+            + if accs.registered_chain.gas_price > 0 {
+                //gas_price[wei, 18 decimals] * EVM_GAS_COST / GWEI = target_native[gwei, 9 decimals]
+                mul_div(accs.registered_chain.gas_price, EVM_GAS_COST, GWEI)
+            } else {
+                0
+            };
```

Comment:
Special casing 0 here here seems redundant
```suggestion
            + mul_div(accs.registered_chain.gas_price, EVM_GAS_COST, GWEI)
```
as `mul_div` will correctly handle that anyway.

Reply:
Yes, this is partly accidental, partly intentional.
Originally it was in there because it prevented a division by zero. That's not the case anymore but I figured I'd leave it in because I assumed that [u128 divisions can still be very expensive](https://github.com/solana-labs/solana/issues/19549) and so while, in the interest of time, I didn't want to bother mucking around with pulling another crate in and getting it to work with better datatypes, but at the same time I figured I can at least avoid the cost.

The check should at least be moved into `mul_div` though.